### PR TITLE
feat: use embedded rest server for handling copilot requests

### PR DESCRIPTION
**Pull request by MarcinVaadin**
## Description

Use embedded rest service instead of plain socket communication.

Kept both for backwards compatibility

### CODE REVIEW

File: src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotRestService.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        return RestUtil.getServiceName()
+    }
+
+    override fun execute(
+        urlDecoder: QueryStringDecoder,
+        request: FullHttpRequest,
+        context: ChannelHandlerContext
+    ): String? {
+        val copilotRequest: CommandRequest = jacksonObjectMapper()
+            .readValue(request.content().toString(Charset.defaultCharset()))
```
Comment (Artur-) [inline src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotRestService.kt:34]: "default charset" always sounds wrong. How will you know which one to use in the request?

File: src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotRestService.kt [hunk 2]
```diff
@@ -30,8 +27,9 @@ class CopilotRestService : RestService() {
         request: FullHttpRequest,
         context: ChannelHandlerContext
     ): String? {
+        val charset = HttpUtil.getCharset(request, Charset.defaultCharset())
         val copilotRequest: CommandRequest = jacksonObjectMapper()
-            .readValue(request.content().toString(Charset.defaultCharset()))
+            .readValue(request.content().toString(charset))
```
Comment (Artur-) [inline src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotRestService.kt:32]: Should we not always just use utf8 everywhere?

**Review by abdullahtellioglu — COMMENTED** [review]
Why not boolean flag for endpoint instead of exposing the url? instead of `endpoint=http\://127.0.0.1\:63342/api/copilot` to `endpoint=true`

**Comment by MarcinVaadin** [comment]
> Why not boolean flag for endpoint instead of exposing the url? instead of `endpoint=http\://127.0.0.1\:63342/api/copilot` to `endpoint=true`

Because `/api/` is forced by IntelliJ, probably for vscode it can be any different, and I don't know for Eclipse. Using full endpoint makes it easier.

File: src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotRestService.kt (hunk 2 above)
Comment (MarcinVaadin) [reply src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotRestService.kt:32]: Currently it looks for charset in request header, then looks for used in system and at last it fallbacks to utf-8.

**Review by abdullahtellioglu — APPROVED** [review]

**Pull request merged**